### ISSUE TITLE
Failing build on trusty - objects_Transport?

### ISSUE THREAD

**Issue by drnic** [Microsoft/service-fabric#98]
I'm trying to ./runbuild.sh on ubuntu trusty (custom changes at https://github.com/Microsoft/service-fabric/compare/master...drnic:trusty?expand=1#files) and I'm getting a build error at:

```
[ 60%] Building CXX object src/ServiceModel/lib/CMakeFiles/objects_servicemodel.dir/__/management/ResourceMonitor/ResourceUsage.cpp.o
[ 60%] Building CXX object src/ServiceModel/lib/CMakeFiles/objects_servicemodel.dir/__/management/ResourceMonitor/ResourceUsageReport.cpp.o
[ 60%] Building CXX object src/ServiceModel/lib/CMakeFiles/objects_servicemodel.dir/__/management/ResourceMonitor/ResourceMeasurementRequest.cpp.o
[ 60%] Building CXX object src/ServiceModel/lib/CMakeFiles/objects_servicemodel.dir/__/management/ResourceMonitor/ResourceMeasurementResponse.cpp.o
[ 60%] Building CXX object src/ServiceModel/lib/CMakeFiles/objects_servicemodel.dir/__/management/ResourceMonitor/ResourceMeasurement.cpp.o
[ 60%] Built target objects_servicemodel
make: *** [all] Error 2
Error: cmd failed: make
```

I'm using dotnet 2.0.0 and a modified trusty 14.04 docker image. What might be causing the build error at this point? Any way to get more verbose errors?

**Comment by drnic** [comment]
I see errors like:
```
[ 15%] Building CXX object src/Transport/lib/CMakeFiles/objects_Transport.dir/__/ServerAuthHeader.cpp.o
/src/prod/src/Common/CryptoUtility.Linux.cpp:1912:27: error: use of undeclared identifier 'X509_check_host'
            auto retval = X509_check_host(cert, matchedName.c_str(), matchedName.size(), 0, nullptr);
                          ^
[ 17%] Built target objects_replication
[ 17%] Building CXX object src/Transport/lib/CMakeFiles/objects_Transport.dir/__/stdafx.cpp.o
1 error generated.
make[2]: *** [src/Common/lib/CMakeFiles/objects_Common.dir/__/CryptoUtility.Linux.cpp.o] Error 1
make[1]: *** [src/Common/lib/CMakeFiles/objects_Common.dir/all] Error 2
[ 17%] Building CXX object src/Transport/lib/CMakeFiles/objects_Transport.dir/__/TcpBufferFactory.cpp.o
[ 19%] Built target objects_HealthManager
[ 19%] Building CXX object src/Transport/lib/CMakeFiles/objects_Transport.dir/__/TcpConnection.cpp.o
/src/prod/src/Transport/SecurityContextSsl.cpp:537:18: error: use of undeclared identifier 'SSL_get0_param'
    auto param = SSL_get0_param(ssl_.get());
                 ^
[ 21%] Built target objects_store
[ 21%] Building CXX object src/Transport/lib/CMakeFiles/objects_Transport.dir/__/TcpConnectionState.cpp.o
```
and later
```
1 error generated.
make[2]: *** [src/Transport/lib/CMakeFiles/objects_Transport.dir/__/SecurityContextSsl.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```
and later
```
make[1]: *** [src/Transport/lib/CMakeFiles/objects_Transport.dir/all] Error 2
```

**Comment by drnic** [comment]
```
/src/prod/src/Common/CryptoUtility.Linux.cpp:1912:27: error: use of undeclared identifier 'X509_check_host'
            auto retval = X509_check_host(cert, matchedName.c_str(), matchedName.size(), 0, nullptr);
                          ^
1 error generated.
make[2]: *** [src/Common/lib/CMakeFiles/objects_Common.dir/__/CryptoUtility.Linux.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 16%] Building CXX object src/Store/lib/CMakeFiles/objects_store.dir/__/ReplicatedStoreSettings.cpp.o
make[1]: *** [src/Common/lib/CMakeFiles/objects_Common.dir/all] Error 2
```

**Comment by drnic** [comment]
Investigating openssl version disparities between 14.04 + 16.04 https://packages.ubuntu.com/search?keywords=openssl and looking to upgrade for trusty

**Comment by bpm-ms** [comment]
@drnic -- Are you targeting 14.04 so that you can generate a debian for 14.04, say for a server or something that hasn't been updated yet?
@anmolah @chenxumsft Do either of you happen to know what differences there are in 14.04 vs 16.04 that might cause the build to fail in 14.04?

**Comment by drnic** [comment]
@wayneeseguin when you arrive (and we have a two-man SF meetup) perhaps you can help find where the missing headers/libraries are coming from in the xenial version vs trusty version

**Comment by drnic** [comment]
I've decided to abandon the effort to backport service fabric to trusty. Primarily because of the current dependency on APIs only available in openssl 1.0.2.

The effort to bring openssl to trusty leaks out into all the other packages that assume openssl 1.0.1. Its messy. More messy than its worth cleaning up.

**Comment by drnic** [comment]
Example of API method available in 1.0.2 https://www.openssl.org/docs/man1.0.2/crypto/X509_check_host.html but not available in 1.0.1 https://www.openssl.org/docs/man1.0.1/crypto/X509_check_host.html (404s)
